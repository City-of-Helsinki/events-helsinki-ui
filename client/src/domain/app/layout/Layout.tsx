import React, { FunctionComponent } from "react";

import Footer from "../footer/Footer";
import Header from "../header/Header";
import Container from "./Container";
import styles from "./layout.module.scss";

const PageLayout: FunctionComponent = ({ children }) => {
  return (
    <div className={styles.pageWrapper}>
      <Header />

      <div className={styles.pageBody}>
        <Container>{children}</Container>
      </div>

      <Footer />
    </div>
  );
};

export default PageLayout;
