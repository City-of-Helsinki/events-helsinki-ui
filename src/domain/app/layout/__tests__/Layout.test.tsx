import { shallow } from "enzyme";
import React from "react";
import { MemoryRouter } from "react-router";

import Layout from "../Layout";

it("Layout matched snapshot", () => {
  const layout = shallow(
    <MemoryRouter>
      <Layout>
        <></>
      </Layout>
    </MemoryRouter>
  );
  expect(layout.html()).toMatchSnapshot();
});
