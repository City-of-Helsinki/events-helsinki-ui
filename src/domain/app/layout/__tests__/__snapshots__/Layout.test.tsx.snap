// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout matched snapshot 1`] = `"<div class=\\"pageWrapper\\"><header class=\\"headerWrapper\\"><div class=\\"container\\"><div class=\\"navbarTop\\"><div class=\\"logoWrapper\\"><div class=\\"logo\\"></div><h1 class=\\"appName\\">Tapahtumat</h1></div><div class=\\"linkWrapper\\"><div class=\\"iconLink\\"><a href=\\"/fi/events\\"><div class=\\"iconWrapper\\"><svg>search.svg</svg></div><div class=\\"textWrapper\\">Etsi tekemistä</div></a></div><div class=\\"iconLink\\"><a href=\\"/fi/collections\\"><div class=\\"iconWrapper\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" class=\\"\\"><path d=\\"M12.013717,1.01899281 C18.0697842,1.02004796 22.9794245,5.92968825 22.9794245,11.9852278 C22.9794245,18.0407674 18.0682014,22.9504077 12.013717,22.9504077 C5.95923261,22.9504077 1.04853717,18.041295 1.04853717,11.9852278 C1.04853717,5.92916067 5.95923261,1.02004796 12.013717,1.01899281 L12.013717,1.01899281 Z M12.013717,2.84652278 C6.97534772,2.84652278 2.87764988,6.94685851 2.87764988,11.9852278 C2.87764988,17.0235971 6.97640288,21.1228777 12.0152998,21.1228777 C17.0541966,21.1228777 21.1534772,17.0235971 21.1534772,11.9852278 C21.1534772,6.94685851 17.0526139,2.84652278 12.013717,2.84652278 Z M16.3688729,14.1029257 L18.0117506,14.903789 C16.9025481,17.205879 14.5690751,18.6656081 12.013717,18.6559233 C9.45683429,18.6671944 7.12141723,17.2069429 6.01199041,14.9032614 L7.65486811,14.1029257 C8.46605189,15.7702798 10.157662,16.828478 12.0118705,16.828478 C13.866079,16.828478 15.5576891,15.7702798 16.3688729,14.1029257 Z M8.0542446,8.33016787 C8.89543918,8.33016787 9.57736211,9.0120908 9.57736211,9.85328537 C9.57736211,10.6944799 8.89543918,11.3764029 8.0542446,11.3764029 C7.21305003,11.3764029 6.5311271,10.6944799 6.5311271,9.85328537 C6.5311271,9.0120908 7.21305003,8.33016787 8.0542446,8.33016787 Z M15.973717,8.33016787 C16.8149116,8.33016787 17.4968345,9.0120908 17.4968345,9.85328537 C17.4968345,10.6944799 16.8149116,11.3764029 15.973717,11.3764029 C15.1325225,11.3764029 14.4505995,10.6944799 14.4505995,9.85328537 C14.4505995,9.0120908 15.1325225,8.33016787 15.973717,8.33016787 Z\\"></path></svg></div><div class=\\"textWrapper\\">Selaa kokoelmia</div></a></div></div><div class=\\"languageDropdown\\"><div class=\\"languageDropdownWrapper\\"><div class=\\"select css-2b097c-container\\"><div class=\\"select__control css-yk16xz-control\\"><div class=\\"select__value-container select__value-container--has-value css-1hwfws3\\"><div class=\\"select__single-value css-1uccc91-singleValue\\">FI</div><input id=\\"react-select-2-input\\" readonly=\\"\\" tabindex=\\"0\\" value=\\"\\" class=\\"css-62g3xt-dummyInput\\"/></div><div class=\\"select__indicators css-1hb7zxy-IndicatorsContainer\\"><div class=\\"select__dropdown-indicator\\"><svg>angle-down.svg</svg></div></div></div></div></div></div></div></div></header><div class=\\"mobileNavigation\\"><header class=\\"mobileHeaderWrapper\\"><div class=\\"container\\"><div class=\\"mobileNavbar\\"><div class=\\"logoWrapper\\"><div class=\\"logo\\"></div><h1 class=\\"appName\\">Tapahtumat</h1></div><div class=\\"buttonWrapper\\"><button class=\\"menuButton\\" aria-label=\\"Avaa valikko\\"><svg>menu.svg</svg></button></div></div></div></header><div class=\\"mobileNavigationMenu\\"><div class=\\"menuItem searchCategories\\"><div class=\\"text\\">Selaa kategorioita</div><div class=\\"iconWrapper\\"><svg>tags-search.svg</svg></div></div><div class=\\"menuItem searchEvents\\"><div class=\\"text\\">Etsi tekemistä</div><div class=\\"iconWrapper\\"><svg>one-eye-smiley.svg</svg></div></div><div class=\\"menuItem mostPopular\\"><div class=\\"text\\">Suosituimmat haut</div><div class=\\"iconWrapper\\"><svg>ranking-stars-ribbon.svg</svg></div></div><div class=\\"menuItem aboutService\\"><div class=\\"text\\">Tietoa palvelusta</div><div class=\\"iconWrapper\\"><svg>flag-information.svg</svg></div></div><div class=\\"login\\"><div class=\\"squareExtension\\"></div></div></div></div><div class=\\"pageBody\\"></div><footer><footer class=\\"topFooterWrapper\\"><div class=\\"container\\"><div class=\\"companyInfoWrapper\\"><div class=\\"logoWrapper\\"><div class=\\"helsinkiLogo\\"></div><h2 class=\\"appName\\">Tapahtumat</h2></div><div class=\\"iconLinkWrapper\\"><div class=\\"iconLink\\"><a href=\\"/fi/events\\"><div class=\\"iconWrapper\\"><svg>search.svg</svg></div><div class=\\"textWrapper\\">Etsi tekemistä</div></a></div><div class=\\"iconLink\\"><a href=\\"/fi/collections\\"><div class=\\"iconWrapper\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" class=\\"\\"><path d=\\"M12.013717,1.01899281 C18.0697842,1.02004796 22.9794245,5.92968825 22.9794245,11.9852278 C22.9794245,18.0407674 18.0682014,22.9504077 12.013717,22.9504077 C5.95923261,22.9504077 1.04853717,18.041295 1.04853717,11.9852278 C1.04853717,5.92916067 5.95923261,1.02004796 12.013717,1.01899281 L12.013717,1.01899281 Z M12.013717,2.84652278 C6.97534772,2.84652278 2.87764988,6.94685851 2.87764988,11.9852278 C2.87764988,17.0235971 6.97640288,21.1228777 12.0152998,21.1228777 C17.0541966,21.1228777 21.1534772,17.0235971 21.1534772,11.9852278 C21.1534772,6.94685851 17.0526139,2.84652278 12.013717,2.84652278 Z M16.3688729,14.1029257 L18.0117506,14.903789 C16.9025481,17.205879 14.5690751,18.6656081 12.013717,18.6559233 C9.45683429,18.6671944 7.12141723,17.2069429 6.01199041,14.9032614 L7.65486811,14.1029257 C8.46605189,15.7702798 10.157662,16.828478 12.0118705,16.828478 C13.866079,16.828478 15.5576891,15.7702798 16.3688729,14.1029257 Z M8.0542446,8.33016787 C8.89543918,8.33016787 9.57736211,9.0120908 9.57736211,9.85328537 C9.57736211,10.6944799 8.89543918,11.3764029 8.0542446,11.3764029 C7.21305003,11.3764029 6.5311271,10.6944799 6.5311271,9.85328537 C6.5311271,9.0120908 7.21305003,8.33016787 8.0542446,8.33016787 Z M15.973717,8.33016787 C16.8149116,8.33016787 17.4968345,9.0120908 17.4968345,9.85328537 C17.4968345,10.6944799 16.8149116,11.3764029 15.973717,11.3764029 C15.1325225,11.3764029 14.4505995,10.6944799 14.4505995,9.85328537 C14.4505995,9.0120908 15.1325225,8.33016787 15.973717,8.33016787 Z\\"></path></svg></div><div class=\\"textWrapper\\">Selaa kokoelmia</div></a></div></div></div><div class=\\"categoriesWrapper\\"><h2 class=\\"categoriesTitle\\">Kategoriat</h2><div class=\\"categoryFilters\\"><div class=\\"categoryFilter withHorizontalPadding\\"><button><svg>movie.svg</svg><span>Elokuva</span><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 199.404 199.404\\" class=\\"\\"><g><polygon points=\\"63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  \\"></polygon></g></svg></button></div><div class=\\"categoryFilter withHorizontalPadding\\"><button><svg>music.svg</svg><span>Musiikki</span><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 199.404 199.404\\" class=\\"\\"><g><polygon points=\\"63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  \\"></polygon></g></svg></button></div><div class=\\"categoryFilter withHorizontalPadding\\"><button><svg>sport.svg</svg><span>Liikunta ja urheilu</span><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 199.404 199.404\\" class=\\"\\"><g><polygon points=\\"63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  \\"></polygon></g></svg></button></div><div class=\\"categoryFilter withHorizontalPadding\\"><button><svg>museum.svg</svg><span>Museot</span><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 199.404 199.404\\" class=\\"\\"><g><polygon points=\\"63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  \\"></polygon></g></svg></button></div><div class=\\"categoryFilter withHorizontalPadding\\"><button><svg>dance.svg</svg><span>Tanssi</span><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 199.404 199.404\\" class=\\"\\"><g><polygon points=\\"63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  \\"></polygon></g></svg></button></div><div class=\\"categoryFilter withHorizontalPadding\\"><button><svg>culture.svg</svg><span>Kulttuuri ja taide</span><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 199.404 199.404\\" class=\\"\\"><g><polygon points=\\"63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  \\"></polygon></g></svg></button></div><div class=\\"categoryFilter withHorizontalPadding\\"><button><svg>nature.svg</svg><span>Luonto ja ulkoilu</span><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 199.404 199.404\\" class=\\"\\"><g><polygon points=\\"63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  \\"></polygon></g></svg></button></div><div class=\\"categoryFilter withHorizontalPadding\\"><button><svg>influence.svg</svg><span>Osallistu ja vaikuta</span><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 199.404 199.404\\" class=\\"\\"><g><polygon points=\\"63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  \\"></polygon></g></svg></button></div><div class=\\"categoryFilter withHorizontalPadding\\"><button><svg>theatre.svg</svg><span>Teatteri</span><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 199.404 199.404\\" class=\\"\\"><g><polygon points=\\"63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  \\"></polygon></g></svg></button></div><div class=\\"categoryFilter withHorizontalPadding\\"><button><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" class=\\"\\"><path d=\\"M13.9714253,10.9514872 C14.1495802,10.7554905 14.4021203,10.6437617 14.6669574,10.6437617 L15.7581797,10.6437617 C16.2917391,10.6437617 16.8001765,10.4170603 17.1567126,10.0201253 L21,5.74180501 L20.4183844,5.16018943 L17.5935184,7.98505543 L17.1781003,7.56963729 L19.9198676,4.66167257 L19.3382897,4.08009472 L16.4303627,6.821862 L16.0149068,6.40648158 L18.8397728,3.58161558 L18.258195,3 L13.9798747,6.84324964 C13.5829397,7.19982348 13.3562006,7.70826089 13.3562006,8.24182028 L13.3562006,9.4018829 C13.3562006,9.66513571 13.2458298,9.91628017 13.0519453,10.0943219 L11.7379445,11.309155 L3.64251505,3.18784912 C3.39091794,2.93621428 2.96044924,3.14269743 3.00292275,3.49606501 C3.24086497,5.47538366 4.30553191,7.76171113 6.54153997,9.99771918 L8.18653998,11.6427569 L9.95854562,11.6427569 L10.5907069,12.3698613 L4.03439548,18.4314657 L5.65114268,20.0338036 L11.6041869,13.5355065 L17.3328306,20.1243333 L18.9900521,18.5844608 L12.7310553,12.305472 L13.9714253,10.9514872 Z\\"></path></svg><span>Ruoka</span><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 199.404 199.404\\" class=\\"\\"><g><polygon points=\\"63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  \\"></polygon></g></svg></button></div></div></div></div></footer><div class=\\"bottomFooter\\"><div class=\\"container\\"><div class=\\"contentWrapper\\"><div class=\\"linkContainer\\"><a href=\\"/about\\">Tietoa palvelusta</a><a href=\\"/faq\\">Usein kysyttyä</a><a href=\\"/feedback\\">Anna palautetta</a></div><div class=\\"logoWrapper\\"><div class=\\"helsinkiLogo\\"></div></div></div></div></div></footer></div>"`;
